### PR TITLE
Add scoring param to SGLCV

### LINKED ISSUE #67: Expose scoring to SGLCV and LogisticSGLCV
#### Describe the workflow you want to enable

`sgl_scoring_path` and `logistic_sgl_scoring_path` have a `scoring` parameter to allow the user to specify an alternative scoring metric. But `SGLCV` and `LogisticSGLCV` do not expose this parameter to the user. We should fix that.

### PR DESCRIPTION
**Pull request by richford**
and add an example for F1 scoring in Logistic SGLCV.
Resolves #67

### CODE REVIEW

File: groupyr/sgl.py [hunk 1]
```diff
@@ -1171,7 +1181,7 @@ def fit(self, X, y):
                 verbose=self.verbose,
                 random_state=self.random_state,
                 return_train_score=True,
-                scoring="neg_mean_squared_error",
+                scoring=self.scoring,
```
Comment (arokem) [inline groupyr/sgl.py:1184]: Just making sure that I understand correctly: the default for BayesSearchCV (when `scoring=None` is passed here) is `"neg_mean_squared_error"`?

File: groupyr/sgl.py [hunk 2]
```diff
@@ -778,6 +778,14 @@ class SGLCV(LinearModel, RegressorMixin, TransformerMixin):
     tol : float, default=1e-7
         Stopping criterion. Convergence tolerance for the ``copt`` proximal gradient solver
 
+    scoring : callable, default=None
+        A string (see sklearn model evaluation documentation) or a scorer
+        callable object / function with signature ``scorer(estimator, X, y)``.
+        For a list of scoring functions that can be used, look at
+        `sklearn.metrics`.  The default scoring option for model selection is
+        "neg_mean_squared_error".  The default scoring option for the ``score``
+        method is "r2".
```
Comment (arokem) [inline groupyr/sgl.py:787]: This is a bit confusing, isn't it? If the user doesn't pass anything here, they will get this behavior ("neg_mean_squared_error" for model selection, "r2" for the `score` method), but if they pass "neg_mean_squared_error" (or any value, really), then this would apply to both the model selection scoring as well as to what they would get out of the `score` method. Should we have two separate scoring inputs? One for model selection and the other for "scoring"?

**Review by arokem — COMMENTED** [review]
This is an improvement! But I think that the SGLCV initialization API might cause some forms of confusion.

File: groupyr/sgl.py (hunk 1 above)
Comment (richford) [reply groupyr/sgl.py:1184]: Hmm, I was incorrect. If `None` is passed, the `score` method of the estimator is used. So at the least, I need to update the docstring, which will resolve the potential confusion you mentioned in your other comment.

But I also think some more testing is necessary to confirm that this actually works. I seem to remember hard-coding "neg_mean_squared_error" because of some issues we were having with "r2".

Let's put a pin in this while I add some tests.

Comment (richford) [reply groupyr/sgl.py:1182]: Okay, I checked and everything still works if I make `r2_score` the default scoring param, which is consistent with default sklearn scoring for regressors. I have done so in the latest commit.

File: groupyr/sgl.py (hunk 2 above)
Comment (richford) [reply groupyr/sgl.py]: Resolved in the latest commit, where I make "r2" the default scoring method for both model selection and the `score` method.
